contracts/src/Verification.sol

```diff
@@ -133,8 +136,17 @@ library Verification {
     {
         for (uint256 i = 0; i < header.digestItems.length; i++) {
             DigestItem memory item = header.digestItems[i];
-            if (item.kind == DIGEST_ITEM_OTHER && item.data.length == 32 && commitment == bytes32(item.data)) {
-                return true;
+            if (item.kind == DIGEST_ITEM_OTHER && item.data.length == 33 && item.data[0] == DIGEST_ITEM_PREFIX) {
+                bytes memory commitment_in_header = new bytes(32);
+                for (uint256 j = 1; j < 33;) {
```

Comment:
Can't we use array slices here? https://docs.soliditylang.org/en/v0.6.0/types.html#array-slices

Reply:
> As of now, array slices are only implemented for calldata arrays. 

As noted on the above page, or error is raised as follows:
`Error (1227): Index range access is only supported for dynamic calldata arrays.`

Reply:
Right, so isn't `header.digestItems[i].data` exactly that?

Reply:
If we remove this line, and reference the calldata directly, then it should compile?
```
 DigestItem memory item = header.digestItems[i];
```

Reply:
https://github.com/Snowfork/snowbridge/pull/1053/commits/7ba6c0ec0a3cd78d038578db1849bf9f604ab06e